socket.cpp

```diff
@@ -56,7 +56,10 @@ bool plain_socket::set_hostname(const char* sAddr)
 	sAddrMb[ln] = '\0';
 
 	if ((sTmp = strstr(sAddrMb, "//")) != nullptr)
+	{
+		sTmp += 2;
```

Comment:
I think you might be reading the code wrong. In case the sAddrMb = "x//\0", sTmp = "//\0", and after addition, sTmp = "\0". strlen(sTmp) = 0, so the args to memmove are

memmove(sAddrMb, sTmp ( = "\0"), 1), so sAddrMb = "\0"

There is no need to check for being behind \0 because the searched for string has a fixed length.